source/blog/2021-08-09-opal-1-2.html.md

```diff
@@ -0,0 +1,178 @@
+
+_An interesting thing is that you can still run older versions of Ruby on the backend, but use the new Ruby 3.0 features when writing Opal code for your frontend._
+
+#### End-less method definition (Opal 1.1)
+
+No more writing code like this:
+
+```ruby
+def ten
+  10
+end
+```
+
+Write it in one line!
+
+```ruby
+def ten = 10
+```
+
+#### Beginless and endless ranges (Opal 1.2)
+
+While those were introduced in earlier Ruby versions (2.6, 2.7), we didn't have those, and now we do! You can absolutely call `a[1..]` instead of `a[1..-1]` now!
+
+#### Numblocks (Opal 1.1)
+
+Also a Ruby 2.7 feature. Personally, I don't like this syntax. Nevertheless, you can use it in Opal now!
+
+```ruby
+[1,2,3].map { _1 ** 2 }
+```
+
+#### Argument forwarding support `def a(...); b(...); end` (Opal 1.1)
+
+We support 3.0 level of argument forwarding. No more writing code like
+
+```ruby
+def example(arg1, arg2, *args, **kwargs, &block)
+  @arg1, @arg2 = arg1, arg2
+  Klass.new(:begin, *args, **kwargs, &block)
+end
+```
+
+...when you can write it like this:
+
+```ruby
+def example(arg1, arg2, ...)
+  @arg1, @arg2 = arg1, arg2
+  Klass.new(:begin, ...)
+end
+```
+
+#### Pattern matching (Opal 1.2)
+
+We support 3.0 level of pattern matching. All kinds of patterns have been fully implemented, including the find pattern.
+
+Pattern matching is optional, if you want to use it, you will need to add `require "corelib/pattern_matching"` to your application.
+
+_Do note, that the find pattern (`[1, 2, 3, "needle", 4] => [*, String => a, *]` to _find_ String and save it to a local variable `a`) and one-line `1 in a` syntax are deemed experimental in Ruby 3.0 and may change their behavior in the future_
+
+#### Methods
+
+There are quite a lot of new methods, some of which got introduced in Ruby 3.0, some earlier.
+
+Opal 1.2:
+- `{Random,SecureRandom}#{hex,base64,urlsafe_base64,uuid,random_float,random_number,alphanumeric}`
+- `String#{+@,-@}`
+- `Hash#except!`
+- `{Array,Hash,Struct}#{deconstruct,deconstruct_keys}`
+
+Opal 1.1:
+- `Array#{difference,intersection,union}`
+- `Enumerable#{filter_map,tally}`
+- `Kernel#then`
+- `{Proc,Method}#{<<,>>}`
+- `Integer#to_d`
+
+#### Breaking changes
+
+Since Opal 1.2, we now follow the Ruby 3.0 semantics for `Array#{to_a,slice,[],uniq,*,difference,-,intersection,&,union,|}` and therefore those don't return an Array subclass anymore, but simply an Array.
+
+#### What isn't there
+
+[We are still using keyword arguments in the Ruby 2.6 style](https://github.com/opal/opal/issues/2241). We want the future releases to warn on the incompatibilities, just like Ruby 2.7 did, to provide a smooth migration curve.
+
+Ractor isn't implemented and likely won't be anytime soon. JavaScript has a very different threading model to Ruby.
+
+### ObjectSpace finalizer support and ObjectSpace::WeakMap implementation (Opal 1.2)
+
+A destructor in Ruby? And also in a niche JavaScript Ruby implementation? Yes!
```

Comment:
niche → nice?

Reply:
I meant niche, as in, little-known-about, but if you happen to have some better sounding alternative, feel free to change it :D